analysis/dep_reg_dataset_jaidip.py

```diff
@@ -0,0 +1,61 @@
+from ehrql import create_dataset
+from ehrql.codes import codelist_from_csv
+
+from ehrql.tables.tpp import (
+    clinical_events,
+    patients,
+    practice_registrations,
+)
+
+dataset = create_dataset()
+qs_start_date = '2024-04-01' # Quality Service Start Date
+min_date = '2006-04-01' # Have their latest first or new episode of depression on or after 1 st April 2006.
+
+# import codelists
+dep_codelist = codelist_from_csv(
+    'codelists/nhsd-primary-care-domain-refsets-depr_cod.csv',
+    column='code')
+depres_codelist = codelist_from_csv(
+    'codelists/nhsd-primary-care-domain-refsets-depres_cod.csv',
+    column='code')
+
+# inclusion criteria
+has_registration = (practice_registrations
+                    .for_patient_on(qs_start_date)
+                    .exists_for_patient())
+
+'''
+Create disease variables
+- DEPR_DAT = Latest date of depression diagnosis
+- DEPRES_DAT = Latest depression resolved date
+- PAT_AGE = Age at quality service start date
+'''
+dataset.DEPR_DAT = ((clinical_events.
```

Comment:
in Python, using all caps for an object name, such as `dataset.DEPR_DAT`, is generally a convention to indicate that the object is a constant. here we want to add a new column to a dataset so would use lower case:

```suggestion
dataset.depr_dat = ((clinical_events.
```